src/lib/utils/functions/queryUtils.ts

```diff
@@ -119,3 +124,28 @@ export function removeLockedColumnFromFacetData(
     return data
   }
 }
+
+/**
+ * Returns true iff the query has filters applied that can be reset.
+ * This includes facet filters and additional filters that are not applied to a locked column.
+ */
+export function hasResettableFilters(
```

Comment:
Ideally I'd rather compare an original Query object with a current Query object to catch if the SQL changes, but the UI isn't set up for that kind of a change